packages/core/lib/segments/attributes/subsegment.d.ts

```diff
@@ -0,0 +1,57 @@
+import * as http from 'http';
+
+declare class Subsegment {
+  id: string;
+  name: string;
+  start_time: number;
+  in_progress?: boolean;
+  counter: number;
+
+  constructor(name: string);
+
+  addNewSubsegment(name: string): Subsegment;
+
+  addSubsegment(subsegment: Subsegment): void;
+
+  removeSubsegment(subsegment: Subsegment): void;
+
+  addAttribute(name: string, data: any): void;
+
+  addPrecursorId(id: string): void;
+
+  addAnnotation(key: string, value: boolean | string | number): void;
+
+  addMetadata(key: string, value: any, namespace?: string): void;
+
+  addSqlData(sqlData: any): void;
+
+  addError(err: Error | string, remote?: boolean): void;
+
+  addRemoteRequestData(req: http.ClientRequest, res: http.IncomingMessage, downstreamXRayEnabled: boolean): void;
```

Comment:
the implementation sets `this.http = new RemoteRequestData(...)`, should this class have a public `http: RemoteRequestData`? I hadn't created a remote_request_data.d.ts yet.

Reply:
Same as above - no need to include it.